### PR TITLE
Port the deploy stage from the development branch

### PR DESCRIPTION
**Pull request by odarbelaeze**
Let's try to enable the 0.12.4 deploy through travis.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,16 @@
...
 .eggs
+.vscode
 
-# produced by coverage.py
+# files created by coverage
+.cache
+.pytest_cache
 .coverage
+
+# Files created by RPN tests
```
Comment (ltalirz) [inline .gitignore]: I think these two lines are not needed here (there isn't even a .ci folder)

File: .travis.yml [hunk 2]
```diff
@@ -67,3 +67,31 @@ after_success:
...
+            - echo "We'll hook up some things in the future"
+          deploy: &pypi
+              provider: pypi
+              skip_existing: true
+              username: aiida-bot
+              password:
+                  secure: NhV11e1kVJaumNMb7YGENngZwa6qJjzoGWzmzqdU5BwFgpXaxbaFAk2fn+Cckrkz4MmNJaW1HaKEAhn07WsbrkjOXK7QVNK48/eagiquKbCQbyNZNjzF+C24EYQHI93y40JQcl2JaCahSqKXbYQCpLlX0Rbtob4psQH29uHRcbq4lm5t3Jmb8WckEzcDLMZX3+uPFwDJxMbbsDW+pONGF1z2/B0j+7m4beTgCepuvIEaXS97rTQj2egKYEJV+V3DbH2o2nr0+3z4lzH2FAdoAnZMlFwHfWoY3WIuYcHcwXDWyLGWQKvncaoh1sLU7gfjjup3dZ4Iq74Zp43x2eXUhbmZBp2cPN3CZpN9baE4NE6MNKeQ/erkg31qdq50OG/rjGgPKyfg5ShO3Kt1CyVbSJX5dqPxtDQblYE8TAGhHfO2O8M1kLhyWvrV5TaQuVB/IAsTkRC+t1NtdWTxaU6wdMz36O5so89oKbFljjk9744m4Ei8DCgc98WH6b96qn0VifjxGuBs6o1aqRoa8O7e29a0TwDVAwxkczgjxbXkP6okRY5IAxsr5Rsbm8urXEo874uQA8n1gGyN6+YKSbjvPbHD9RIRl9IddC6y2wKTv/1JBifNjHcLhadl7RVRPSU7vFSs2cknN+5Gkw3FgZ/jA8SoMxMLFsvHcFxchrurAkE=
+              on:
+                  repo: aiidateam/aiida_core
+                  all_branches: true
```
Comment (ltalirz) [inline .travis.yml]: I see, so this will be active on all branches?
Is that the result of your discussion with @sphuber  ?

Comment (odarbelaeze) [reply .travis.yml]: This is correct.

**Comment by odarbelaeze** [comment]
```
required_version = StrictVersion(PIP_REQUIRED_VERSION)
    installed_version = StrictVersion(pip.__version__)

    if installed_version < required_version:
        print 'The installation requires pip>={}, whereas currently {} is installed'.format(required_version, installed_version)
        sys.exit(1)

    bin_folder = path.join(aiida_folder, 'bin')
```

No, there's no problem building aiida with the 19.x version of `pip`. I just tried in a fresh machine:

```
$ docker run --rm -it -v $PWD:/code -w /code python:2.7 /bin/bash
root@947c08f49da0:/code# pip --version
pip 19.1 from /usr/local/lib/python2.7/site-packages/pip (python 2.7)
root@947c08f49da0:/code# python setup.py sdist
fastentrypoints
/usr/local/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'reentry_register'
  warnings.warn(msg)
running sdist
running egg_info
writing requirements to aiida_core.egg-info/requires.txt
[...]
Writing aiida-core-0.12.3/setup.cfg
Creating tar archive
removing 'aiida-core-0.12.3' (and everything under it)
root@947c08f49da0:/code# 
```

**Pull request merged**